File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
"""
prices = [7,1,5,3,6,4]

cur_low = 1
max_profit = 5


"""
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        cur_low = float("inf")
        max_profit = 0
        
        for price in prices:
            if price < cur_low:
                cur_low = price
            else:
                max_profit = max(max_profit, price - cur_low)
        
        return max_profit
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cs
/*
maxProfit = 0
minPrice = prices[i]



*/

public class Solution {
    public int MaxProfit(int[] prices) {
        
        int maxProfit = 0;
        int minPrice = prices[0];
        
        foreach (int price in prices){
            if (price < minPrice){
                minPrice = price;
            }
            else if (price - minPrice > maxProfit) {
             maxProfit = price - minPrice;   
            }
        }
        return maxProfit;
    }
}